api/incubator/src/main/java/io/opentelemetry/api/incubator/internal/InternalExtendedAttributeKeyImpl.java

```diff
@@ -0,0 +1,178 @@
+   */
+  @Nullable
+  public static <T> AttributeKey<T> toAttributeKey(ExtendedAttributeKey<T> extendedAttributeKey) {
+    switch (extendedAttributeKey.getType()) {
+      case STRING:
+        return InternalAttributeKeyImpl.create(extendedAttributeKey.getKey(), AttributeType.STRING);
+      case BOOLEAN:
+        return InternalAttributeKeyImpl.create(
+            extendedAttributeKey.getKey(), AttributeType.BOOLEAN);
+      case LONG:
+        return InternalAttributeKeyImpl.create(extendedAttributeKey.getKey(), AttributeType.LONG);
+      case DOUBLE:
+        return InternalAttributeKeyImpl.create(extendedAttributeKey.getKey(), AttributeType.DOUBLE);
+      case STRING_ARRAY:
+        return InternalAttributeKeyImpl.create(
+            extendedAttributeKey.getKey(), AttributeType.STRING_ARRAY);
+      case BOOLEAN_ARRAY:
+        return InternalAttributeKeyImpl.create(
+            extendedAttributeKey.getKey(), AttributeType.BOOLEAN_ARRAY);
+      case LONG_ARRAY:
+        return InternalAttributeKeyImpl.create(
+            extendedAttributeKey.getKey(), AttributeType.LONG_ARRAY);
+      case DOUBLE_ARRAY:
+        return InternalAttributeKeyImpl.create(
+            extendedAttributeKey.getKey(), AttributeType.DOUBLE_ARRAY);
+      case EXTENDED_ATTRIBUTES:
+        return null;
+    }
+    throw new IllegalArgumentException(
+        "Unrecognized extendedAttributeKey type: " + extendedAttributeKey.getType());
+  }
+
+  /** Return the equivalent {@link ExtendedAttributeKey} for the {@link AttributeKey}. */
+  public static <T> ExtendedAttributeKey<T> toExtendedAttributeKey(AttributeKey<T> attributeKey) {
+    switch (attributeKey.getType()) {
+      case STRING:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.STRING);
+      case BOOLEAN:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.BOOLEAN);
+      case LONG:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.LONG);
+      case DOUBLE:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.DOUBLE);
+      case STRING_ARRAY:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.STRING_ARRAY);
+      case BOOLEAN_ARRAY:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.BOOLEAN_ARRAY);
+      case LONG_ARRAY:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.LONG_ARRAY);
+      case DOUBLE_ARRAY:
+        return InternalExtendedAttributeKeyImpl.create(
+            attributeKey.getKey(), ExtendedAttributeType.DOUBLE_ARRAY);
+    }
+    throw new IllegalArgumentException("Unrecognized attributeKey type: " + attributeKey.getType());
```

Comment:
Seems like a worthy place to add test coverage.

Reply:
This is the fall through case of a switch on an enum. Reaching this is a programming error, since the switch cases should be updated in lockstep when any enum entries are added.